runtime/doc/api.txt

```diff
@@ -2753,6 +2753,8 @@ nvim_buf_set_extmark({buffer}, {ns_id}, {line}, {col}, {*opts})
                   • priority: a priority value for the highlight group or sign
                     attribute. For example treesitter highlighting uses a
                     value of 100.
+                  • subpriority: a priority value to determine tiebreaks
```

Comment:
I think there could be value in exposing it even to plugin authors. I can imagine a plugin that wants to use a fixed `priority` (or that can be set by the user) to make the plugin's highlight priority relative to e.g. treesitter or LSP, but may also want to enforce individual extmark priorities within the plugin.

I wouldn't be opposed to adding some scary text like "Most users should not need to set this" or "(Advanced)" or something though.

Reply:
Ok, I'll let it be your call.

I am one to be (overly) conservative when it comes to adding API surface area so I would personally prefer we wait for a concrete request first instead of exposing this early for potential use cases. It's a decent use case though so the cost/benefit is probably justified with either decision.